### PR TITLE
circulation: resource creation for ILL request

### PR DESCRIPTION
**Pull request by zannkukai**
This commit creates the `ill_request` resource, implements permissions
and basics tests.

Co-Authored-by: Renaud Michotte <renaud.michotte@gmail.com>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?

### CODE REVIEW

File: rero_ils/modules/ill_requests/cli.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+from rero_ils.modules.ill_requests.api import ILLRequest
+from rero_ils.modules.items.cli import get_locations
+from rero_ils.modules.locations.api import Location
+from rero_ils.modules.patrons.api import Patron
+from rero_ils.modules.utils import get_ref_for_pid
+
+
+@click.command('create_ill_requests')
+@click.option('-f', '--requests_f', 'request_f', help='Request input file.')
```
Comment (lauren-d) [inline rero_ils/modules/ill_requests/cli.py:36]: cosmetic: maybe better to use `request-file` instead `requests_f` and `request_f`

File: scripts/setup [hunk 2]
```diff
@@ -367,6 +371,8 @@ eval ${PREFIX} invenio utils runindex --raise-on-error
 info_msg "Serials patterns: ${DATA_PATH}/patterns.json ${CREATE_LAZY} ${DONT_STOP}"
 eval ${PREFIX} invenio fixtures create_patterns ${DATA_PATH}/patterns.json
 
+
+
```
Comment (lauren-d) [inline scripts/setup:375]: can be removed?

File: rero_ils/modules/ill_requests/views.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+from __future__ import absolute_import, print_function
+
+from flask import Blueprint
+
+blueprint = Blueprint(
+    'ill_requests',
+    __name__,
+    template_folder='templates',
+    static_folder='static',
+)
```
Comment (lauren-d) [inline rero_ils/modules/ill_requests/views.py:29]: Is really needed?

Comment (zannkukai) [reply rero_ils/modules/ill_requests/views.py:29]: It will be in next PR's (for managing the WTForms REST API)

File: data/ill_requests.json [hunk 4]
```diff
@@ -0,0 +1,81 @@
+[
+  {
+    "status": "pending",
+    "copy": false,
+    "document": {
+      "title": "Histoire d'\u00c9gypte",
+      "authors": "Aly Badr, Cairo University",
+      "publisher": "Cairo University Press",
+      "year": "1920",
```
Comment (BadrAly) [inline data/ill_requests.json:9]: I am not that old :-)

File: rero_ils/modules/ill_requests/cli.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+from rero_ils.modules.utils import get_ref_for_pid
+
+
+@click.command('create_ill_requests')
+@click.option('-f', '--requests_file', 'input_file', help='Request input file.')
+@with_appcontext
+def create_ill_requests(input_file):
+    """Create ILL request for each organisation."""
+    location_pids = get_locations()
+    patron_pids = list(Patron.get_all_pids())
```
Comment (BadrAly) [inline rero_ils/modules/ill_requests/cli.py:41]: I think this  way you will allow ill_requests from another ogranisations. I do not know if it is the case in this US or not @iGormilhit ?
If not, I you think you need to filter patrons by organsation as well.

**Review by BadrAly — APPROVED** [review]

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message title proposition: "circulation: resource creation for ILL request"

File: data/ill_requests.json (hunk 4 above)
Comment (zannkukai) [reply data/ill_requests.json:9]: It's not me ; these data come from UserStory fixtures ;-)

File: rero_ils/modules/ill_requests/cli.py (hunk 5 above)
Comment (iGormilhit) [reply rero_ils/modules/ill_requests/cli.py:41]: @BadrAly @zannkukai If a patron from org A needs a document from another org, he or she needs to create an ILL request as if it were a non RERO ILS network, at least for now. So I think Aly is right.

**Review by lauren-d — APPROVED** [review]

File: rero_ils/modules/ill_requests/jsonschemas/ill_requests/ill_request-v0.0.1.json [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+        "templateOptions": {
+          "selectWithSortOptions": {
+            "order": "label"
+          }
+        }
+      }
+    },
+    "copy": {
+      "title": "Copy",
+      "description": "Is this request for a copy request or not?",
```
Comment (iGormilhit) [inline rero_ils/modules/ill_requests/jsonschemas/ill_requests/ill_request-v0.0.1.json:80]: I think that my proposition wasn't so good, maybe it's better like this:
```suggestion
      "description": "Define if the request is for a copy or an actual document.",
```

**Review by iGormilhit — CHANGES_REQUESTED** [review]

File: rero_ils/modules/ill_requests/jsonschemas/ill_requests/ill_request-v0.0.1.json (hunk 6 above)
Comment (zannkukai) [reply rero_ils/modules/ill_requests/jsonschemas/ill_requests/ill_request-v0.0.1.json:80]: done

**Review by benerken — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]

**Pull request merged**